FakeNFT/Scenes/FavouriteNft/FavouriteNftInteractor.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+
+protocol FavouriteNftInteractorProtocol: AnyObject {
+    func loadFavouriteNft()
+    func removeFromFavourite(_ favouriteNft: Nft)
+}
+
+final class FavouriteNftInteractor: FavouriteNftInteractorProtocol {
+    weak var presenter: FavouriteNftPresenterProtocol?
+    
+    private let profile: Profile
+    private let nftService: NftService
+    
+    init(profile: Profile, nftService: NftService) {
+        self.profile = profile
+        self.nftService = nftService
+    }
+    
+    func loadFavouriteNft() {
+        var nfts = [Nft]()
+        let favouriteNftIds = profile.likes
+        let dispatchGroup = DispatchGroup()
+        var encounteredError: Error?
+        
+        for nftId in favouriteNftIds {
+            dispatchGroup.enter()
+            nftService.loadNft(id: nftId) { result in
+                switch result {
+                case .success(let nft):
+                    nfts.append(nft)
+                case .failure(let error):
+                    encounteredError = error
+                }
+                dispatchGroup.leave()
+            }
+        }
```

Comment:
поскольку тут один запрос выполняется, то нет особого смысла юзать диспатч группу, можно вызов методов презентера передать сразу в замыкание

```suggestion
    func loadFavouriteNft() {
        let favouriteNftIds = profile.likes
        
        for nftId in favouriteNftIds {
            nftService.loadNft(id: nftId) { [weak self] result in
                guard let self else { return }
                switch result {
                case .success(let nft):
                    self.presenter?.didFetchFavouriteNfts([nft])
                case .failure(let error):
                    self.presenter?.didFailFetchFavouriteNfts(error: error)
                }
            }
        }
    }
```

Reply:
Отличный комментарий, это нужно исправить.

Reply:
Тут выполняется несколько запросов на получение nft, посколько likes это массив ид nft. DispatchGroup тут позволяет загрузить все nft и разом вызвать метод у presenter